### PR TITLE
Improve GitHub support

### PR DESCRIPTION
**Pull request by sa2ajj**

### CODE REVIEW

File: master/buildbot/status/github.py [hunk 1]
```diff
@@ -14,52 +14,96 @@
...
+    """
+    Convert Buildbot states into GitHub states.
+    """
+    # GitHub defines `success`, `failure` and `error` states.
+    # We explicitly map success and failure. Any other BuildBot status
+    # is converted to `error`.
+    return _STATE_MAP.get(results, 'error')
+
+
+def _timeDeltaToHumanReadable(start, end):
```
Comment (tardyp) [inline master/buildbot/status/github.py]: There is nothing specific to Github here.
That could be used by other status plugins.

Comment (sa2ajj) [inline master/buildbot/status/github.py]: True.

That was my next step (first I wanted to make sure that there's a test for custom GitHub URL).

**Comment by sa2ajj** [comment]
I still do not have test cases for the new functionalit

**Comment by sa2ajj** [comment]
And, actually, I am not sure how to do that.  It seems that `FakeRequest` needs to be re-written (current implementation is really hackish).

**Comment by djmitche** [comment]
Yes, FakeRequest is pretty bogus.  Twisted's testing utilities may have a useful alternative.

**Comment by sa2ajj** [comment]
@djmitche I looked at what Twisted offers and it's not much.

Maybe you could suggest what test cases would be "enough"?  I really want this PR to land in eight before next release.

**Comment by djmitche** [comment]
This looks like a good set of tests now.  Coverage is only missing for the various error-handling bits (`raise ValueError..`) and for the signature verification.

**Comment by sa2ajj** [comment]
- [ ] signature verification
- [ ] error cases

**Comment by sa2ajj** [comment]
(Needs port to `master`)

**Comment by djmitche** [comment]
I still get a few pep8 changes and some test failures from `validate.sh`.
```
[ERROR]
Traceback (most recent call last):
  File "/home/dustin/code/buildbot/t/buildbot/master/buildbot/test/unit/test_status_github.py", line 70, in setUp
    token='token', repoOwner='owner', repoName='name')
  File "/home/dustin/code/buildbot/t/buildbot/master/buildbot/status/github.py", line 77, in __init__
    self._github = GitHubAPI(oauth2_token=token, baseURL=baseURL)
exceptions.TypeError: __init__() got an unexpected keyword argument 'baseURL'

buildbot.test.unit.test_status_github.TestGitHubStatus.test_buildFinished_failure
buildbot.test.unit.test_status_github.TestGitHubStatus.test_buildFinished_success
buildbot.test.unit.test_status_github.TestGitHubStatus.test_buildStarted_failure
buildbot.test.unit.test_status_github.TestGitHubStatus.test_buildStarted_success
buildbot.test.unit.test_status_github.TestGitHubStatus.test_builderAdded
buildbot.test.unit.test_status_github.TestGitHubStatus.test_custom_github_url
buildbot.test.unit.test_status_github.TestGitHubStatus.test_getGitHubRepoProperties_ok
buildbot.test.unit.test_status_github.TestGitHubStatus.test_getGitHubRepoProperties_skip_no_name
buildbot.test.unit.test_status_github.TestGitHubStatus.test_getGitHubRepoProperties_skip_no_owner
buildbot.test.unit.test_status_github.TestGitHubStatus.test_getGitHubRepoProperties_skip_no_sha
buildbot.test.unit.test_status_github.TestGitHubStatus.test_getGitHubState
buildbot.test.unit.test_status_github.TestGitHubStatus.test_initialization_required_arguments
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendFinishStatus_failure
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendFinishStatus_no_properties
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendFinishStatus_ok
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendGitHubStatus_error
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendGitHubStatus_success
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendStartStatus_failure
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendStartStatus_no_properties
buildbot.test.unit.test_status_github.TestGitHubStatus.test_sendStartStatus_ok
buildbot.test.unit.test_status_github.TestGitHubStatus.test_startService
```

**Comment by sa2ajj** [comment]
What is the exact command you are running?

**Comment by djmitche** [comment]
`sh common/validate.sh HEAD^`

**Comment by djmitche** [comment]
and, I did re-run `pip install -e master` in case of `setup.py` differences between master and eight.

**Comment by sa2ajj** [comment]
Right.  I believe you need to update txgithub package (since it's OK to run Buildbot without it, txgithub package does not have a proper versioned dependency).

**Comment by djmitche** [comment]
OK, looks good to me!  Sorry to delay this so much..

**Pull request merged**